eks-public-cluster.tf

```diff
@@ -0,0 +1,135 @@
+# Define a KMS main key to encrypt the EKS cluster
+resource "aws_kms_key" "eks-public" {
+  description         = "EKS Secret Encryption Key"
+  enable_key_rotation = true
+}
+
+# EKS Cluster definition
+module "eks-public" {
+  source       = "terraform-aws-modules/eks/aws"
+  version      = "18.29.0"
+  cluster_name = local.public_cluster_name
+  # From https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html
+  cluster_version = var.kubernetes_version
+  subnet_ids      = module.vpc.public_subnets
```

Comment:
Is it correct?
That's the point I'm not sure, reusing the existing public subnet or recreating a new one.

Reply:
that looks fine, you are correct: we want this cluster to have private IPs for its pods and Kube-services in these subnets.

Why multiple subnets? Because a given VPC spans across different availability zones.